### PR TITLE
feat: enhance okta-sdk oauth to include client secret

### PR DESCRIPTION
**Pull request by Astro2024**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our [guidelines](/okta/okta-sdk-nodejs/blob/master/CONTRIBUTING.md)
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Adding Tests
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation changes
- [ ] Other... Please describe:

## What is the current behavior?
Currently, when using the okta-sdk there are only two ways to get an access token for the Okta client. The first is using an access key and the second is using a public/private key pair.

Issue Number: N/A

## What is the new behavior?
This PR adds the ability to retrieve an access token for the Okta client using client id and client secrets.

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -20,9 +20,10 @@
     "jest": "JEST_JUNIT_OUTPUT_DIR=./test-reports jest --coverage --ci --testResultsProcessor=jest-junit test/jest/*.js",
     "predocs": "rimraf ./jsdocs && mkdir jsdocs/ && ./utils/make-jsdoc-readme.js > ./jsdocs/jsdoc-temp.md",
     "docs": "./node_modules/.bin/jsdoc src/ -c ./docs/config.json -d ./jsdocs/ -P ./package.json -R ./jsdocs/jsdoc-temp.md -r",
-    "test:integration": "yarn test:integration:oauth && yarn test:integration:ssws",
+    "test:integration": "yarn test:integration:oauth && yarn test:integration:ssws && && yarn test:integration:clientsecret",
```
Comment (willpots) [inline package.json]:
```suggestion
    "test:integration": "yarn test:integration:oauth && yarn test:integration:ssws && yarn test:integration:clientsecret",
```
intentional?

File: package.json [hunk 2]
```diff
@@ -20,9 +20,10 @@
...
     "test:integration:ssws": "TEST_TYPE=it OKTA_CLIENT_AUTHORIZATIONMODE=SSWS  mocha test/it/*.ts",
     "test:integration:oauth": "TEST_TYPE=it OKTA_CLIENT_AUTHORIZATIONMODE=PrivateKey  mocha test/it/user-get.ts",
+    "test:integration:clientsecret": "TEST_TYPE=it OKTA_CLIENT_AUTHORIZATIONMODE=ClientSecret  mocha test/it/user-get.ts",
```
Comment (willpots) [inline package.json:26]:
```suggestion
    "test:integration:clientsecret": "TEST_TYPE=it OKTA_CLIENT_AUTHORIZATIONMODE=ClientSecret mocha test/it/user-get.ts",
```

**Review by willpots — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (Astro2024) [reply package.json]: Nope

File: package.json (hunk 2 above)
Comment (Astro2024) [reply package.json:26]: I added the extra space because all of the other commands have it. I figured it was a weird styling thing

**Pull request merged**